### PR TITLE
Encode the project name with space

### PR DESCRIPTION
**Pull request by Priya5**
This is for encoding project name having space.
Reference: https://scan.coverity.com/projects/ohloh-ui

### CODE REVIEW

**Comment by vapier** [comment]
i assume you've tested this completely ?

there are 4 places this is used:
* twice via `curl --data`
  * i thought curl takes care of encoding this, but i guess not.  using `+` here is correct afaict: \
https://www.w3.org/html/wg/spec/association-of-controls-and-forms.html#url-encoded-form-data
* once via `curl --form`
  * i thought curl takes care of encoding this, but i guess not.  using `+` here is incorrect afaict: \
https://www.w3.org/html/wg/spec/association-of-controls-and-forms.html#multipart-form-data
although it seems that using `%2F` is also wrong.  i wonder if coverity servers are handling these ?
  * the current "submit build" form uses `enctype="multipart/form-data"` (which is why `curl --form`) with `accept-charset="UTF-8"` which means we can pass the project directly, but i think we'd have to update the other form options too.
* once as a URL query string
  * using `+` here is incorrect in general afaict: \
https://en.wikipedia.org/wiki/Percent-encoding#Reserved_characters
i wonder if coverity servers support this specifically ?

**Comment by Priya5** [comment]
I have updated the PR to use percent encoding of spaces. 

- Coverity servers supports both `+` and `%20`.
- `%2F` is handling only `/`. The action fails while submitting the build for project name with spaces. Reference - https://github.com/blackducksoftware/ohloh-ui/actions/runs/3289839743/jobs/5421872353
- Verified it manually.
```
curl --form token=token  --form email=email   --form file=@ohloh-ui_2022_06_08_30.tgz  --form version="Version" --form description="Description" "https://scan.coverity.com/builds?project=Ohloh UI"
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>400 Bad Request</title>
</head><body>
<h1>Bad Request</h1>
<p>Your browser sent a request that this server could not understand.<br />
</p>
<hr>
<address>Apache Server at scan.coverity.com Port 80</address>
```
```
curl --form token=token  --form email=email   --form file=@ohloh-ui_2022_06_08_30.tgz  --form version="Version" --form description="Description" "https://scan.coverity.com/builds?project=Ohloh+UI"
Build successfully submitted.
```
```
curl --form token=token  --form email=email   --form file=@ohloh-ui_2022_06_08_30.tgz  --form version="Version" --form description="Description" "https://scan.coverity.com/builds?project=Ohloh%20UI"
Build successfully submitted.
```

**Comment by vapier** [comment]
> once via `curl --form`

i looked at the code & site again and i think this is a (very) old hold over from pre-GH action days.  i deleted this usage in commit 69efeec2e8db2e30da7a0b62146411018a324d5b.  which leaves the other 3.

> once as a URL query string

i've seen other services handle `+` as a space, so i'd believe Coverity does the same.

File: action.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ runs:
     # here.  Wait to see if anyone needs something else.
```
Comment (vapier) [inline action.yml]: please update the comment here to reflect these new changes

**Comment by vapier** [comment]
thanks for the fix!

**Pull request merged**